\hypertarget{classErrorException}{}\section{Error\+Exception Class Reference}
\label{classErrorException}\index{Error\+Exception@{Error\+Exception}}


~\newline
 This exception is thrown by calls to the {\ttfamily error} function.  




{\ttfamily \#include $<$error.\+h$>$}

Inheritance diagram for Error\+Exception\+:\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[height=2.000000cm]{classErrorException}
\end{center}
\end{figure}
\subsection*{Public Member Functions}
\begin{DoxyCompactItemize}
\item 
\mbox{\hyperlink{classErrorException_a4f422b01fff7cc16eea81e00638f44d2}{Error\+Exception}} (std\+::string msg)
\begin{DoxyCompactList}\small\item\em Creates a new \mbox{\hyperlink{classErrorException}{Error\+Exception}} with the given error message. \end{DoxyCompactList}\item 
virtual \mbox{\hyperlink{classErrorException_a61c8f6114a93ea2ff11a5f08e3640d76}{$\sim$\+Error\+Exception}} () noexcept=default
\begin{DoxyCompactList}\small\item\em Frees any memory allocated by the exception. \end{DoxyCompactList}\item 
virtual void \mbox{\hyperlink{classErrorException_af1e6a5fb477a95a876e86ceadf065ffd}{dump}} () const
\begin{DoxyCompactList}\small\item\em Prints the exception to the standard error stream (cerr), including its message and stack trace if any. \end{DoxyCompactList}\item 
virtual void \mbox{\hyperlink{classErrorException_a277bc85222ba9a7d0fd47fd10d44d115}{dump}} (std\+::ostream \&out) const
\begin{DoxyCompactList}\small\item\em Prints the exception to the given output stream, including its message and stack trace if any. \end{DoxyCompactList}\item 
virtual std\+::string \mbox{\hyperlink{classErrorException_ae6faf7d90a6394517c0328748c99ec1d}{get\+Kind}} () const
\begin{DoxyCompactList}\small\item\em Returns what kind of exception this is. \end{DoxyCompactList}\item 
virtual std\+::string \mbox{\hyperlink{classErrorException_a1c1cc72e6e4257dbd29ff04a23973008}{get\+Message}} () const
\begin{DoxyCompactList}\small\item\em Returns the exception\textquotesingle{}s error message as passed to its constructor. \end{DoxyCompactList}\item 
virtual std\+::string \mbox{\hyperlink{classErrorException_a79ec2353dbf71c4867c65c6c6df5dfce}{get\+Stack\+Trace}} () const
\begin{DoxyCompactList}\small\item\em Returns a stack trace for this exception as a multi-\/line string. \end{DoxyCompactList}\item 
virtual bool \mbox{\hyperlink{classErrorException_ae7ccac41c27ba02a30bdec527db0a639}{has\+Stack\+Trace}} () const
\begin{DoxyCompactList}\small\item\em Returns whether this exception has a non-\/empty stack trace. \end{DoxyCompactList}\item 
void \mbox{\hyperlink{classErrorException_a896ae13e5a5b1b89f058f7de9697c3cd}{set\+Kind}} (const std\+::string \&kind)
\begin{DoxyCompactList}\small\item\em Sets what kind of exception this is. \end{DoxyCompactList}\item 
virtual const char $\ast$ \mbox{\hyperlink{classErrorException_a576a441c2b1e2f01d271e237b314eac3}{what}} () const noexcept
\begin{DoxyCompactList}\small\item\em Returns the exception\textquotesingle{}s error message as a C string. \end{DoxyCompactList}\end{DoxyCompactItemize}
\subsection*{Protected Member Functions}
\begin{DoxyCompactItemize}
\item 
void \mbox{\hyperlink{classErrorException_aaac5126664da146aa7290235ac7b1d07}{set\+Stack\+Trace}} (const std\+::string \&stack\+Trace)
\begin{DoxyCompactList}\small\item\em Sets this exception\textquotesingle{}s stack trace to the given multi-\/line string. \end{DoxyCompactList}\end{DoxyCompactItemize}


\subsection{Detailed Description}
~\newline
 This exception is thrown by calls to the {\ttfamily error} function. 

Typical code for catching errors looks like this\+:


\begin{DoxyPre}
     try \{
        ... code in which an error might occur ...
     \} catch (\mbox{\hyperlink{classErrorException}{ErrorException}}\& ex) \{
        ... code to handle the error condition ...
     \}
*\end{DoxyPre}


If an {\ttfamily \mbox{\hyperlink{classErrorException}{Error\+Exception}}} is thrown at any point in the range of the {\ttfamily try} (including in functions called from that code), control will jump immediately to the error handler. 

\subsection{Constructor \& Destructor Documentation}
\mbox{\Hypertarget{classErrorException_a4f422b01fff7cc16eea81e00638f44d2}\label{classErrorException_a4f422b01fff7cc16eea81e00638f44d2}} 
\index{Error\+Exception@{Error\+Exception}!Error\+Exception@{Error\+Exception}}
\index{Error\+Exception@{Error\+Exception}!Error\+Exception@{Error\+Exception}}
\subsubsection{\texorpdfstring{Error\+Exception()}{ErrorException()}}
{\footnotesize\ttfamily \mbox{\hyperlink{classErrorException}{Error\+Exception}} (\begin{DoxyParamCaption}\item[{std\+::string}]{msg }\end{DoxyParamCaption})}



Creates a new \mbox{\hyperlink{classErrorException}{Error\+Exception}} with the given error message. 

\mbox{\Hypertarget{classErrorException_a61c8f6114a93ea2ff11a5f08e3640d76}\label{classErrorException_a61c8f6114a93ea2ff11a5f08e3640d76}} 
\index{Error\+Exception@{Error\+Exception}!````~Error\+Exception@{$\sim$\+Error\+Exception}}
\index{````~Error\+Exception@{$\sim$\+Error\+Exception}!Error\+Exception@{Error\+Exception}}
\subsubsection{\texorpdfstring{$\sim$\+Error\+Exception()}{~ErrorException()}}
{\footnotesize\ttfamily virtual $\sim$\mbox{\hyperlink{classErrorException}{Error\+Exception}} (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [virtual]}, {\ttfamily [default]}, {\ttfamily [noexcept]}}



Frees any memory allocated by the exception. 



\subsection{Member Function Documentation}
\mbox{\Hypertarget{classErrorException_af1e6a5fb477a95a876e86ceadf065ffd}\label{classErrorException_af1e6a5fb477a95a876e86ceadf065ffd}} 
\index{Error\+Exception@{Error\+Exception}!dump@{dump}}
\index{dump@{dump}!Error\+Exception@{Error\+Exception}}
\subsubsection{\texorpdfstring{dump()}{dump()}\hspace{0.1cm}{\footnotesize\ttfamily [1/2]}}
{\footnotesize\ttfamily void dump (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption}) const\hspace{0.3cm}{\ttfamily [virtual]}}



Prints the exception to the standard error stream (cerr), including its message and stack trace if any. 

\mbox{\Hypertarget{classErrorException_a277bc85222ba9a7d0fd47fd10d44d115}\label{classErrorException_a277bc85222ba9a7d0fd47fd10d44d115}} 
\index{Error\+Exception@{Error\+Exception}!dump@{dump}}
\index{dump@{dump}!Error\+Exception@{Error\+Exception}}
\subsubsection{\texorpdfstring{dump()}{dump()}\hspace{0.1cm}{\footnotesize\ttfamily [2/2]}}
{\footnotesize\ttfamily void dump (\begin{DoxyParamCaption}\item[{std\+::ostream \&}]{out }\end{DoxyParamCaption}) const\hspace{0.3cm}{\ttfamily [virtual]}}



Prints the exception to the given output stream, including its message and stack trace if any. 

\mbox{\Hypertarget{classErrorException_ae6faf7d90a6394517c0328748c99ec1d}\label{classErrorException_ae6faf7d90a6394517c0328748c99ec1d}} 
\index{Error\+Exception@{Error\+Exception}!get\+Kind@{get\+Kind}}
\index{get\+Kind@{get\+Kind}!Error\+Exception@{Error\+Exception}}
\subsubsection{\texorpdfstring{get\+Kind()}{getKind()}}
{\footnotesize\ttfamily std\+::string get\+Kind (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption}) const\hspace{0.3cm}{\ttfamily [virtual]}}



Returns what kind of exception this is. 

In general this returns \char`\"{}error\char`\"{}, but in some cases we catch other kinds of exceptions (like thrown ints or strings) and wrap them up as Error\+Exceptions. In such cases, the kind will be \char`\"{}int\char`\"{} or \char`\"{}string\char`\"{} etc. \mbox{\Hypertarget{classErrorException_a1c1cc72e6e4257dbd29ff04a23973008}\label{classErrorException_a1c1cc72e6e4257dbd29ff04a23973008}} 
\index{Error\+Exception@{Error\+Exception}!get\+Message@{get\+Message}}
\index{get\+Message@{get\+Message}!Error\+Exception@{Error\+Exception}}
\subsubsection{\texorpdfstring{get\+Message()}{getMessage()}}
{\footnotesize\ttfamily std\+::string get\+Message (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption}) const\hspace{0.3cm}{\ttfamily [virtual]}}



Returns the exception\textquotesingle{}s error message as passed to its constructor. 

\mbox{\Hypertarget{classErrorException_a79ec2353dbf71c4867c65c6c6df5dfce}\label{classErrorException_a79ec2353dbf71c4867c65c6c6df5dfce}} 
\index{Error\+Exception@{Error\+Exception}!get\+Stack\+Trace@{get\+Stack\+Trace}}
\index{get\+Stack\+Trace@{get\+Stack\+Trace}!Error\+Exception@{Error\+Exception}}
\subsubsection{\texorpdfstring{get\+Stack\+Trace()}{getStackTrace()}}
{\footnotesize\ttfamily std\+::string get\+Stack\+Trace (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption}) const\hspace{0.3cm}{\ttfamily [virtual]}}



Returns a stack trace for this exception as a multi-\/line string. 

See \mbox{\hyperlink{exceptions_8h_source}{exceptions.\+h}}/cpp for descriptions of the format. Not every exception has a proper stack trace, based on when/why it was thrown, platform incompatibilities, and other issues; use has\+Stack\+Trace to check if a given exception\textquotesingle{}s stack trace is populated. \mbox{\Hypertarget{classErrorException_ae7ccac41c27ba02a30bdec527db0a639}\label{classErrorException_ae7ccac41c27ba02a30bdec527db0a639}} 
\index{Error\+Exception@{Error\+Exception}!has\+Stack\+Trace@{has\+Stack\+Trace}}
\index{has\+Stack\+Trace@{has\+Stack\+Trace}!Error\+Exception@{Error\+Exception}}
\subsubsection{\texorpdfstring{has\+Stack\+Trace()}{hasStackTrace()}}
{\footnotesize\ttfamily bool has\+Stack\+Trace (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption}) const\hspace{0.3cm}{\ttfamily [virtual]}}



Returns whether this exception has a non-\/empty stack trace. 

Not every exception has a proper stack trace, based on when/why it was thrown, platform incompatibilities, and other issues; use has\+Stack\+Trace to check if a given exception\textquotesingle{}s stack trace is populated. \mbox{\Hypertarget{classErrorException_a896ae13e5a5b1b89f058f7de9697c3cd}\label{classErrorException_a896ae13e5a5b1b89f058f7de9697c3cd}} 
\index{Error\+Exception@{Error\+Exception}!set\+Kind@{set\+Kind}}
\index{set\+Kind@{set\+Kind}!Error\+Exception@{Error\+Exception}}
\subsubsection{\texorpdfstring{set\+Kind()}{setKind()}}
{\footnotesize\ttfamily void set\+Kind (\begin{DoxyParamCaption}\item[{const std\+::string \&}]{kind }\end{DoxyParamCaption})}



Sets what kind of exception this is. 

Default is \char`\"{}error\char`\"{}. \mbox{\Hypertarget{classErrorException_aaac5126664da146aa7290235ac7b1d07}\label{classErrorException_aaac5126664da146aa7290235ac7b1d07}} 
\index{Error\+Exception@{Error\+Exception}!set\+Stack\+Trace@{set\+Stack\+Trace}}
\index{set\+Stack\+Trace@{set\+Stack\+Trace}!Error\+Exception@{Error\+Exception}}
\subsubsection{\texorpdfstring{set\+Stack\+Trace()}{setStackTrace()}}
{\footnotesize\ttfamily void set\+Stack\+Trace (\begin{DoxyParamCaption}\item[{const std\+::string \&}]{stack\+Trace }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Sets this exception\textquotesingle{}s stack trace to the given multi-\/line string. 

\mbox{\Hypertarget{classErrorException_a576a441c2b1e2f01d271e237b314eac3}\label{classErrorException_a576a441c2b1e2f01d271e237b314eac3}} 
\index{Error\+Exception@{Error\+Exception}!what@{what}}
\index{what@{what}!Error\+Exception@{Error\+Exception}}
\subsubsection{\texorpdfstring{what()}{what()}}
{\footnotesize\ttfamily const char $\ast$ what (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption}) const\hspace{0.3cm}{\ttfamily [virtual]}, {\ttfamily [noexcept]}}



Returns the exception\textquotesingle{}s error message as a C string. 

